Sources/CartonHelpers/DiagnosticsParser.swift

```diff
@@ -0,0 +1,259 @@
+                  .note,
+                file: file,
+                line: components[0],
+                char: components[1],
+                code: String(lines[lineIdx]),
+                message: components[3]
+              )
+            )
+          }
+        }
+      } else {
+        terminal.write(String(line) + "\n", inColor: .cyan)
+      }
+      lineIdx += 1
+    }
+    if let currFile = currFile {
+      diagnostics[currFile] = fileMessages
+    }
+
+    outputDiagnostics(diagnostics, terminal)
+  }
+
+  func outputDiagnostics(
+    _ diagnostics: [String.SubSequence: [CustomDiagnostic]],
+    _ terminal: InteractiveWriter
+  ) {
+    for (file, messages) in diagnostics.sorted(by: { $0.key < $1.key }) {
+      guard messages.count > 0 else { continue }
+      terminal.write("\(" \(file) ", color: "[1m", "[7m")") // bold, reversed
+      terminal.write(" \(messages.first!.file)\(messages.first!.line)\n\n", inColor: .grey)
+      // Group messages that occur on sequential lines to provie a more readable output
+      var groupedMessages = [[CustomDiagnostic]]()
+      for message in messages {
+        if let lastLineStr = groupedMessages.last?.last?.line,
+          let lastLine = Int(lastLineStr),
+          let line = Int(message.line),
+          lastLine == line - 1 || lastLine == line
+        {
+          groupedMessages[groupedMessages.count - 1].append(message)
+        } else {
+          groupedMessages.append([message])
+        }
+      }
+      for messages in groupedMessages {
+        // Output the diagnostic message
+        for message in messages {
+          let kind = message.kind.rawValue.uppercased()
+          terminal
+            .write(
+              "  \(" \(kind) ", color: message.kind.color, "[37;1m") \(message.message)\n"
+            ) // 37;1: bright white
+        }
+        let maxLine = messages.map(\.line.count).max() ?? 0
+        for (offset, message) in messages.enumerated() {
+          if offset > 0 {
+            // Make sure we don't log the same line twice
+            if messages[offset - 1].line != message.line {
+              flush(messages: messages, message: message, maxLine: maxLine, terminal)
+            }
+          } else {
+            flush(messages: messages, message: message, maxLine: maxLine, terminal)
+          }
+        }
+        terminal.write("\n")
+      }
+      terminal.write("\n")
+    }
+  }
+
+  func flush(
+    messages: [CustomDiagnostic],
+    message: CustomDiagnostic,
+    maxLine: Int,
+    _ terminal: InteractiveWriter
+  ) {
+    // Get all diagnostics for a particular line.
+    let allChars = messages.filter { $0.line == message.line }.map(\.char)
+    // Output the code for this line, syntax highlighted
+    terminal
+      .write(
+        "  \("\(message.line.padding(toLength: maxLine, withPad: " ", startingAt: 0)) | ", color: "[36m")\(Self.highlighter.highlight(message.code))\n"
```

Comment:
<!--
  1 failure:  Line should be 10...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: Sources/CartonHelpers/DiagnosticsParser.swift;
  Line: 235;
-->
- :no_entry_sign: Line should be 100 characters or less: currently 151 characters (`line_length`)